Scripts/Python/grsnWallPython.py

```diff
@@ -40,1207 +40,1187 @@
+            self.ChangeGameState(kSouth, kSetBlocker)
+            SouthPanelSound.run(self.key, state='up')
+            return
+        ### Tube Entry trigger ###
+        if(id == NorthTubeEntry.id and ageSDL["nState"][0] >= kWait and ageSDL["nState"][0] != kEnd):
+            if ageSDL["nState"][0] == kWait:
+                self.ChangeGameState(kNorth, kEntry)
+            if(ageSDL["nWallPlayer"] == (-1,)):
+                self.ChangeChuteState(kNorth, PtGetClientIDFromAvatarKey(PtFindAvatar(events).getKey()))
+                if(PtFindAvatar(events) == PtGetLocalAvatar() and cID == ageSDL["sWallPlayer"][0]):
+                    ageSDL["sWallPlayer"] = (-1,)
+            return
+        if(id == SouthTubeEntry.id and ageSDL["sState"][0] >= kWait and ageSDL["sState"][0] != kEnd):
+            if ageSDL["sState"][0] == kWait:
+                self.ChangeGameState(kSouth, kEntry)
+            if(ageSDL["sWallPlayer"] == (-1,)):
+                self.ChangeChuteState(kSouth, PtGetClientIDFromAvatarKey(PtFindAvatar(events).getKey()))
+                if(PtFindAvatar(events) == PtGetLocalAvatar() and cID == ageSDL["nWallPlayer"][0]):
+                    ageSDL["nWallPlayer"] = (-1,)
+            return
+        ### Tube Multibehaviors ###
+        if(id == NorthTubeMulti.id):
+            for event in events:
+                if(event[0] == kMultiStageEvent and event[1] == 0 and event[2] == kEnterStage):
+                    #Smart seek complete
+                    NorthTubeClose.run(self.key)
+                    if(PtFindAvatar(events) != PtGetLocalAvatar()):
+                        NorthTubeExclude.clear(self.key)
+                elif(event[0] == kMultiStageEvent and event[1] == 0 and event[2] == kAdvanceNextStage):
+                    #Multistage complete
+                    if(PtFindAvatar(events) == PtGetLocalAvatar()):
+                        avatar = PtGetLocalAvatar()
+                        currentgender = avatar.avatar.getAvatarClothingGroup()
+                        if currentgender == kFemaleClothingGroup:
+                            clothing = FemaleSuit
+                        else:
+                            clothing = MaleSuit
+                        if not wornItem:
+                            wornItem = avatar.avatar.getAvatarClothingList()
+                            for item in wornItem:
+                                DefaultColor1.append(avatar.avatar.getTintClothingItem(item[0],1))
+                                DefaultColor2.append(avatar.avatar.getTintClothingItem(item[0],2))
+                        i = 1
+                        for item in clothing[0:]:
+                            avatar.avatar.netForce(1)
+                            avatar.avatar.wearClothingItem(item, 0)
+                            avatar.avatar.tintClothingItem(item, DefaultColor1[i], 0)
+                            avatar.avatar.tintClothingItemLayer(item, DefaultColor2[i], 2, 1)
+                            i += 1
```

Comment:
What's the reasoning for starting the index at 1 instead of 0 here? Is it possible to rework this area and the South tube multibehavior block below to use `zip` instead of the `i`-indexed loop just like you do in a couple other places like https://github.com/H-uru/Plasma/blob/9f72c65cc5cc424010fdcaa63f4f8f7e8ac140e5/Scripts/Python/grsnWallPython.py#L306-L310

Reply:
1 is used because the wornitem, which we got the colors from initially has the face in the 0 position.
1 is the first item we are wearing.
This is why the clothing was rearranged in https://github.com/H-uru/Plasma/commit/9f72c65cc5cc424010fdcaa63f4f8f7e8ac140e5 , without doing that you would wear the suit and put the wrong colors the wrong clothing slots.